extras/linked-token/src/LinkedTokenController.sol

```diff
@@ -7,15 +7,58 @@ import {
 import {IERC20} from "openzeppelin-contracts/token/ERC20/IERC20.sol";
 import {LinkedToken} from "./LinkedToken.sol";
 import {SubnetID} from "@ipc/src/structs/Subnet.sol";
+import {
+    Initializable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {
+    UUPSUpgradeable
+} from "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
 
-contract LinkedTokenController is LinkedToken {
+contract LinkedTokenController is Initializable, LinkedToken, UUPSUpgradeable {
     using SafeERC20 for IERC20;
 
-    constructor(
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(
+        address gateway,
+        address underlyingToken,
+        SubnetID memory linkedSubnet,
+        address linkedContract
+    ) public initializer {
+        __LinkedToken_init(
+            gateway,
+            underlyingToken,
+            linkedSubnet,
+            linkedContract
+        );
+        __UUPSUpgradeable_init();
+    }
+
+    function reinitialize(
         address gateway,
         address underlyingToken,
-        SubnetID memory linkedSubnet
-    ) LinkedToken(gateway, underlyingToken, linkedSubnet) {}
+        SubnetID memory linkedSubnet,
+        address linkedContract
+    ) public reinitializer(2) {
+        __LinkedToken_init(
+            gateway,
+            underlyingToken,
+            linkedSubnet,
+            linkedContract
+        );
+        __UUPSUpgradeable_init();
+    }
```

Comment:
What's the reason we need this here? My understanding is that reinitializers are used as part of an upgrade to a new implementation? Putting this here will enable this contract to be initialized twice, IUC.

Reply:
fixed in https://github.com/consensus-shipyard/ipc/pull/856